packages/siopv2-oid4vp-rp-rest-client/src/agent/SiopV2OID4VpRpRestClient.ts

```diff
@@ -0,0 +1,96 @@
+import { fetch } from 'cross-fetch'
+import {
+  ISiopV2OID4VpRpRestClient,
+  IGenerateAuthRequestArgs,
+  IGenerateAuthRequestURIResponse,
+  IGetAuthStatusArgs,
+  IRequiredContext,
+  IRemoveAuthRequestSessionArgs,
+} from '../index'
+import Debug from 'debug'
+import { IAgentPlugin } from '@veramo/core'
+
+const debug = Debug('ssi-sdk-siopv2-oid4vp-rp-rest-client:SiopV2OID4VpRpRestClient')
+// todo merge this with Niels's branch feature/siop-verifier and use those classes/definitions
+export class SiopV2OID4VpRpRestClient implements IAgentPlugin {
+  readonly methods: ISiopV2OID4VpRpRestClient = {
+    removeAuthRequestSession: this.removeAuthRequestSession.bind(this),
+    generateAuthRequest: this.generateAuthRequest.bind(this),
+    getAuthStatus: this.getAuthStatus.bind(this),
+  }
+
+  private readonly baseUrl?: string
+  private readonly definitionId?: string
+
+  constructor(baseUrl?: string, definitionId?: string) {
+    if (baseUrl) {
+      this.baseUrl = baseUrl
+    }
+    if (definitionId) {
+      this.definitionId = definitionId
+    }
+  }
+
+  private async removeAuthRequestSession(args: IRemoveAuthRequestSessionArgs, context: IRequiredContext): Promise<void> {
+    const baseUrl = this.checkBaseUrlParameter(args.baseUrl)
+    const definitionId = this.checkDefinitionIdParameter(args.definitionId)
+    fetch(this.uriWithBase(`/webapp/definitions/${definitionId}/auth-requests/${args.correlationId}`, baseUrl), {
+      method: 'DELETE',
+    })
+  }
+
+  private async getAuthStatus(args: IGetAuthStatusArgs, context: IRequiredContext): Promise<any> {
```

Comment:
Why return an any? We have interface for the payloads. This means any implementation is left on their own to figure out what the payload is

Reply:
changed to the actual response object from common